### PR TITLE
Fix #6179: Add clean copy feature

### LINKED ISSUE #6179: "Copy clean URL" feature
### Description: 
Per: https://github.com/brave/brave-browser/issues/23315

### Testing strategy:
Found in the following document: https://docs.google.com/document/d/1ea7eF3s0WTsTlmMPr8LYp5KUyNg8eK4omjizo27qvJE

Implement the "Copy clean URL" feature on iOS

### PR DESCRIPTION
**Pull request by cuba**
## Summary of Changes

This pull request fixes #6179 

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Screenshots:
| Navigation link | Address bar |
|--|--|
| Simulator Screenshot - iPhone 14 Pro - 2023-09-07 at 11 34 02 | Screenshot 2023-09-07 at 11 31 38 AMScreenshot 2023-09-07 at 11 31 04 AM |

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by iccub — APPROVED** [review]
🚀

File: Sources/Brave/Extensions/UIAction+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Strings
+import BraveCore
+
+extension UIAction {
+  static func makePasteAndGoAction(pasteCallback: @escaping (String) -> Void) -> UIAction {
```
Comment (kylehickinson) [inline Sources/Brave/Extensions/UIAction+Extensions.swift]: None of these really belong in a generic extension on `UIAction`

Comment (cuba) [reply Sources/Brave/Extensions/UIAction+Extensions.swift]: I just want to share the code between those two components. Is there any suggestion you may have?

Comment (iccub) [reply Sources/Brave/Extensions/UIAction+Extensions.swift]: A `UrlUIActions` struct maybe? This way you do not expose those actions for every action

Comment (cuba) [reply Sources/Brave/Extensions/UIAction+Extensions.swift]: what if we just prefix brave on this. I mean we do this to colors and other things so we have `Color.bravePrimary` and I think now we will have `Image.braveBroom` (I'm not sure if this is what direction we decided). What's the problem with having `UIAction.bravePasteAndGo(pasteCallback:)`

Comment (kylehickinson) [reply Sources/Brave/Extensions/UIAction+Extensions.swift]: The issues here are: 
1. Low reusability: You're only using this in very few specific spots and the repeated logic isn't worth abstracting away. The paste related actions aren't even reused _at all_.
2. Over exposed: You're exposing these very specific features actions on `UIAction` which is used throughout the front end that possibly have nothing to do with. The paste actions don't even handle any additional logic to make them worth exposing.

You're better off just repeating the (very simple) logic in two places than introduce functions like this against `UIAction`

**Review by kylehickinson — APPROVED** [review]

**Comment by iccub** [comment]
Remember to add a test plan for it later

**Pull request merged**